### DISCUSSION TITLE
Error during parsing. This message / key probably does not conform to a valid OpenPGP format.

### DISCUSSION THREAD

**Discussion by Naziatanveer** [openpgpjs/openpgpjs#1638] [Q&A]
**_Encryption Code:_**

const openpgp = require("openpgp");
const fs = require("fs");

async function encryption() {
  const dataToEncrypt = fs
    .readFileSync("./dataToEncrypt.txt", "utf-8")
    .toString();
  const publicArmored = fs.readFileSync("./public.pem").toString();
  const message = await openpgp.createMessage({ text: dataToEncrypt });
  const publicKey = await openpgp.readKey({ armoredKey: publicArmored });
  const encrypt = openpgp.encrypt({
    message,
    encryptionKeys: publicKey,
    format: "binary",
  });
  return encrypt;
}

encryption().then((result) => {
    console.log("Successful encryption");
    fs.writeFileSync("binaryEncrypted.txt.pgp", result);
  }).catch((error) => {
    console.log("encryptions failed with error " + error.message);
  });

**Result : Encryption in binary format is generated** 
****************************************************************************************************************************
_**Decryption Code:**_

const openpgp = _require("openpgp");_
const fs = require("fs");

async function decryption() {
  const dataToDecrypt = fs.readFileSync("./binaryEncrypted.txt.pgp", "ascii");
  const privateArmored = fs.readFileSync("./private.pem").toString();
  const privateKey = await openpgp.readKey({ armoredKey: privateArmored });
  const unitArray = new Uint8Array(dataToDecrypt);
  const message = await openpgp.readMessage({ binaryMessage: unitArray });

  const decrypt = openpgp.decrypt({
    message,
    decryptionKeys: privateKey,
  });
  return decrypt;
}
decryption()then((result) => {
    console.log("Successful Decryption");
    fs.writeFileSync("binaryDecrypted.txt.pgp", result);
  }).catch((error) => {
    console.log("encryptions failed with error " + error.message);
  });

**Result: Error during parsing. This message / key probably does not conform to a valid OpenPGP format.**

**Can someone please help me why decryption is not working as expected ?**

**Comment by larabr** [comment]
Check that you're reading the file `binaryEncrypted.txt.pgp` correctly -- `'ascii'` does not look like the right option. Other than that, there are other discussions / issues about the same problem, so you can search for those and report if the solutions don't work.
